src/Features/CSharp/Portable/Completion/KeywordRecommenders/ReadOnlyKeywordRecommender.cs

```diff
@@ -43,7 +43,7 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
         private static bool IsRefReadOnlyContext(CSharpSyntaxContext context)
             => context.TargetToken.IsKind(SyntaxKind.RefKeyword) &&
                (context.TargetToken.Parent.IsKind(SyntaxKind.RefType) ||
-                (context.TargetToken.Parent.IsKind(SyntaxKind.Parameter) && context.IsFunctionPointerTypeArgumentContext));
+                (context.TargetToken.Parent.IsKind(SyntaxKind.FunctionPointerParameter) && context.IsFunctionPointerTypeArgumentContext));
```

Comment:
Deleting the IsKind() check doesn't work anymore? I would have assumed context.IsFunctionPointerTypeArgumentContext has already figured this out accurately enough and we don't have to do extra guesses.

Reply:
I think we can remove this. I'll commit and we'll see if the tests pass :)

---
In reply to: [451070402](https://github.com/dotnet/roslyn/pull/45684#discussion_r451070402) [](ancestors = 451070402)